File: hotel/src/main/java/at/fhtw/hotel/domain/model/Room.java
package at.fhtw.hotel.domain.model;

import java.math.BigDecimal;
import java.util.List;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class Room {
  Long id;
  String title;
  String description;
  String imageUrl;
  BigDecimal pricePerNight;
  List<Extra> extras;
}

File: frontend/src/types/room.ts
export interface Extra {
  id: number;
  name: string;
  icon: string;
}

export interface RoomAvailabilitySummary {
  available: boolean;
  checkIn: string | null;
  checkOut: string | null;
  nights: number | null;
  totalPrice: number | null;
}

export interface Room {
  id: number;
  title: string;
  description: string;
  imageUrl: string;
  pricePerNight: number;
  extras: Extra[];
  availability: RoomAvailabilitySummary | null;
}

export interface RoomPage {
  content: Room[];
  totalElements: number;
  totalPages: number;
  currentPage: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}

export interface AvailabilityResponse {
  roomId: number;
  roomTitle: string;
  checkIn: string;
  checkOut: string;
  available: boolean;
  nights: number;
  totalPrice: number;
}
